tools/syntax.txt

```diff
@@ -615,6 +619,16 @@ list<VariableDimensionSyntax> dimensions
 UntypedSyntax base=DataTypeSyntax kind=Untyped
 token untyped
 
+PropertyTypeSyntax base=DataTypeSyntax kind=PropertyType
```

Comment:
Rather than have separate types for these, you could just reuse KeywordTypeSyntax, which represents all types that are a single keyword.